Makefile

```diff
@@ -0,0 +1,157 @@
+mkfile_dir := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+##### Variables used
+# (can be overriden using command line args to make)
+# Paths used in build.
+BUILD_DIR:=$(abspath ${mkfile_dir}/../build)
```

Comment:
should be `BUILD_DIR?=...` if we want to be able to override them using command line args (or env var) right?